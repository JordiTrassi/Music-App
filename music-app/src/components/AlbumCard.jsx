import { Link } from 'react-router-dom';
import { Box, Card, CardContent, CardMedia, IconButton, Tooltip, Typography } from '@mui/material';
import { AddCircle } from '@mui/icons-material';


export const AlbumCard = ({
    artistName,
    artworkUrl100,
    collectionName,
    collectionId,
}) => {



  return (
      <Card
          className='animate__animated animate__fadeIn'
          sx={{ display: 'flex', mb: '20px', justifyContent: 'space-between', boxShadow: 6 }}
      >
      <Box sx={{ display: 'flex', flexDirection: 'column' }} >
        <CardContent sx={{ flex: '1 0 auto' }}>
          <Typography component="div" variant="h5">
            {collectionName}
          </Typography>
          <Typography variant="subtitle1" color="text.secondary" component="div">
            {artistName}
          </Typography>
        </CardContent>
              
        <Box sx={{ display: 'flex', alignItems: 'center', pl: 1, pb: 1 }}>
          <Link to={`/album/${collectionId}`}>
            <Tooltip
              title="more info"
              placement="right"
              arrow
            >
              <IconButton>
                    <AddCircle fontSize='large'/> 
                </IconButton>              
            </Tooltip>
          </Link>
               
        </Box>
      </Box>
        <CardMedia
            component="img"
            image={artworkUrl100}
            alt={collectionName}
            sx={{ height: 200, width: 'auto' }}     
        />
    </Card>
  );
}

