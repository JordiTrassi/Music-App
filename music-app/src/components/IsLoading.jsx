import { CircularProgress } from '@mui/material';
import { Box } from '@mui/material';


export const IsLoading = () => {
  
  return (
    <Box sx={{ display: 'flex', justifyContent: 'center', p:10 }}>
      <CircularProgress />
    </Box>
  );
}
