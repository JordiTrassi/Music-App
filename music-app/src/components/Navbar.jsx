import { AppBar, Box, Toolbar, Typography, InputBase, Link, Tooltip  } from '@mui/material';
import { styled, alpha } from '@mui/material/styles';

import SearchIcon from '@mui/icons-material/Search';
import { MusicNote } from '@mui/icons-material';

const Search = styled('div')(({ theme }) => ({
  position: 'relative',
  borderRadius: theme.shape.borderRadius,
  backgroundColor: alpha(theme.palette.common.white, 0.15),
  '&:hover': {
    backgroundColor: alpha(theme.palette.common.white, 0.25),
  },
  marginLeft: 0,
  width: '100%',
  [theme.breakpoints.up('sm')]: {
    marginLeft: theme.spacing(1),
    width: 'auto',
  },
}));

const SearchIconWrapper = styled('div')(({ theme }) => ({
  padding: theme.spacing(0, 2),
  height: '100%',
  position: 'absolute',
  pointerEvents: 'none',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
}));

const StyledInputBase = styled(InputBase)(({ theme }) => ({
  color: 'inherit',
  '& .MuiInputBase-input': {
    padding: theme.spacing(1, 1, 1, 0),
    // vertical padding + font size from searchIcon
    paddingLeft: `calc(1em + ${theme.spacing(4)})`,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      width: '20ch',
      '&:focus': {
        width: '30ch',
      },
    },
  },
}));



export const Navbar = () => {

  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static">
        <Toolbar>          
          <Typography
                variant="h5"
                noWrap
                component="div"
                sx={{ flexGrow: 1, display: { xs: '2', sm: 'block' } }}
          >
            <Tooltip
              title="home"
              arrow
              placement="bottom"
              
            >
              <Link
                color="inherit"
                underline="none" 
                href="/"
                >
                    <MusicNote sx={{fontSize: 35}} /> MUSIC APP
              </Link>
            </Tooltip>
          </Typography>
            <Search>
                <SearchIconWrapper>
                    <SearchIcon />
                </SearchIconWrapper>
                <StyledInputBase
                placeholder="Search Album or Artist ..."
                inputProps={{ 'aria-label': 'search' }}
                />
            </Search>
        </Toolbar>
      </AppBar>
    </Box>
  );
}

