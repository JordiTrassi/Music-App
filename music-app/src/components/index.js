export * from './AlbumCard';
export * from './Navbar';