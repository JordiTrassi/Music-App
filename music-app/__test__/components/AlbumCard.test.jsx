import { screen, render } from '@testing-library/react';
import { AlbumCard } from '../../src/components/AlbumCard';
import { BrowserRouter as Router } from 'react-router-dom';


describe('Pruebas en el componente <AlbumCard />', () => {
    
    const artistName = 'Elvis'; 
    const artworkUrl100 = 'Elvis_Image.jp';
    const collectionName = 'Elvis Collection';

    test('Debe de hacer match con el snapshot ', () => {
        const { container } = render(
            <Router>
                <AlbumCard
                    artistName={artistName}
                    artworkUrl100={artworkUrl100}
                    collectionName={collectionName}
                />
            </Router>
        );
        
        expect(container).toMatchSnapshot();

    });

    test('Debe de mostrar el "collectionName" en un h5', () => {
        const { getByTestId } = render(
            <Router>
                <AlbumCard
                    artistName={artistName}
                    artworkUrl100={artworkUrl100}
                    collectionName={collectionName}
                />
            </Router>
        );

        expect(getByTestId('test-collectionName').innerHTML).toBe(collectionName);
    });
    
    test('Debe de mostrar el "artistName"', () => {
        render(
            <Router>
                <AlbumCard
                    artistName={artistName}
                    artworkUrl100={artworkUrl100}
                    collectionName={collectionName}
                />
            </Router>
        );
       
        expect(screen.getByText(artistName)).toBeTruthy();
    });


})
