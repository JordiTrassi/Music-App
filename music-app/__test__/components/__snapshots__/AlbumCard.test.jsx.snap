// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AlbumCard /> Debe de hacer match con el snapshot  1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root animate__animated animate__fadeIn css-1bz0hi6-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-j7qwjs"
    >
      <div
        class="MuiCardContent-root css-1qaznj3-MuiCardContent-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          data-testid="test-collectionName"
        >
          Elvis Collection
        </div>
        <div
          class="MuiTypography-root MuiTypography-subtitle1 css-ecpxie-MuiTypography-root"
        >
          Elvis
        </div>
      </div>
      <div
        class="MuiBox-root css-1ndgitk"
      >
        <a
          href="/album/undefined"
        >
          <button
            aria-label="more info"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
              data-testid="AddCircleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
    <img
      alt="Elvis Collection"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1c4t542-MuiCardMedia-root"
      src="Elvis_Image.jp"
    />
  </div>
</div>
`;
